### PR TITLE
[Doc]Add model artifact metadata information to the core concept page

### PR DESCRIPTION
**Pull request by yubozhao**
## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [x] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: docs/source/concepts.rst [hunk 1]
```diff
@@ -352,6 +352,71 @@ models are depending on each other, such as the example above.
...
+            'precision_score': 0.792,
+            'mean_absolute_error': 0.88
+        }
+    )
+    svc.save()
+
+.. note::
+
+    The model artifact metadata is immutable.
+
```
Comment (jackyzha0) [inline docs/source/concepts.rst:389]: might be more clear to put this above the code block

File: docs/source/concepts.rst [hunk 2]
```diff
@@ -352,6 +352,71 @@ models are depending on each other, such as the example above.
 
 .. _concepts-api-func-and-adapters:
 
+Model Artifact Metadata
+^^^^^^^^^^^^^^^^^^^^^^^
+
+BentoML allows users to include additional metadata information for the packed model
+artifacts. The metadata are intended to be used to specify identifying attributes of
+the model artifact that are meaningful and relevant to users.
```
Comment (jackyzha0) [inline docs/source/concepts.rst:360]:
```suggestion
the model artifact that are meaningful and relevant to users, such as accuracy, dataset used, and other static information.
```

**Review by jackyzha0 — COMMENTED** [review]
oops, should've included this in my PR 😅 thanks!

File: docs/source/concepts.rst (hunk 2 above)
Comment (yubozhao) [reply docs/source/concepts.rst:360]: Good suggestion!

File: docs/source/concepts.rst (hunk 1 above)
Comment (yubozhao) [reply docs/source/concepts.rst:389]: Yeah, might be.  I am trying to catch readers' attention on this.  Maybe have the message in the code is more natural.

**Pull request merged**